src/Persistent/Pure.hs

```diff
@@ -737,11 +737,13 @@ findComment :: CommentKey -> MQuery Comment
 findComment ck = findComment' (ck ^. ckIdeaId) (ck ^. ckParents) (ck ^. ckCommentId)
 
 instance FromProto IdeaLike where
-    fromProto () = IdeaLike
+    fromProto p m = IdeaLike m (_protoIdeaLikeDelegate p)
 
 -- FIXME: Assumption: the given @AUID Idea@ MUST be in the DB.
-addLikeToIdea :: AUID Idea -> AddDb IdeaLike
-addLikeToIdea iid = addDb' (mkIdeaVoteLikeKey iid) (dbIdeaMap . at iid . _Just . ideaLikes)
+addLikeToIdea :: AUID Idea -> User -> AddDb IdeaLike
+addLikeToIdea iid user =
```

Comment:
also, if one of `delegate` or `delegatee` is a good name for something, then `user` probably isn't.